contracts/FleetFactory.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity >= 0.5.0 < 0.7.0;
+
+import "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxyFactory.sol";
+import "@gnosis.pm/safe-contracts/contracts/proxies/GnosisSafeProxy.sol";
+import "@gnosis.pm/safe-contracts/contracts/GnosisSafe.sol";
+
+contract FleetFactory {
+  GnosisSafeProxyFactory public proxyFactory;
+
+  event FleetDeployed(address indexed owner, address[] fleet);
+
+  constructor(GnosisSafeProxyFactory _proxyFactory) public {
+    proxyFactory = _proxyFactory;
+  }
+
+  function deployFleet(address owner, uint256 size) public returns (address[] memory) {
```

Comment:
Why don't we hand over the template address? I think handing over the template address would be more general and the owner could also deploy other safes with other modules than just this current own one?

Reply:
Everything was structured around our own use case: since we don't even have a parameter in our scripts to change the template I thought I could get rid of an extra parameter by retrieving the master copy from the owner. Note that you can specify the template using `deployFleetFromTemplate` however.
I absolutely agree that in the general case the template should be among the input values.